File: 136-single-number/136-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
  let obj = {};
  for (let i = 0; i < nums.length; i++) {
    if (obj.hasOwnProperty(nums[i])) {
      delete obj[nums[i]];
    } else {
      obj[nums[i]] = true
    }
  }

  return +Object.keys(obj)[0]
};

File: 136-single-number/136-single-number.go
import (
	"strconv"
)

func singleNumber(nums []int) int {
	obj := make(map[string]bool)
	result := 0
	for _, n := range nums {
		numberString := strconv.Itoa(n)
		if _, ok := obj[numberString]; ok {
			delete(obj, numberString)
		} else {
			obj[numberString] = true
		}
	}

	for k := range obj {
		result, _ = strconv.Atoi(k)
	}

	return result
}